### PR TITLE
[MISP-2343] Update UBoot environment size and some environment variables

### PR DESCRIPTION
**Pull request by alexborro**
- Update UBoot to MSC version 2020.01
- Change UBoot environment size from 128KiB to 32KiB to comply with the new UBoot version/settings (2020.01)
- Add new UBoot environment variables
- Fix the **make** command for the UBoot 2020.01 (it should make the defconfig before compiling: **make defconfig && make**)
- Move the **-dev** suffix to the most right of the package name (from VERSION-dev-imx6dl to VERSION-imx6dl-dev)
- Enable shellcheck and fix issues.

### CODE REVIEW

**Review by robinmdh — APPROVED** [review]

File: env/u-boot_env.txt [hunk 1]
```diff
@@ -1,70 +1,85 @@
...
-sysinfo=setenv bootargs_sysinfo board_name=${board_name} board_variant=${variant_key}-${feature_key} serial_number=${serial_number} uboot_version=<${uboot_version}> 
-uboot_version=U-Boot 2018.03 (Jul 08 2019 - 13:52:58 +0000)
-usb_boot=if usb start && usb dev 0; then setenv uenvpart ${usbfatpart}; if run usbloadenv; then run importenv; fi; if test -n $uenvcmd; then echo Running uenvcmd ...;run uenvcmd; fi;if run usbloadimage; then run usbboot; else echo ERR: Load image(s) from USB failed; false; fi; else echo ERR: USB start failed; false; fi; 
-usbargs=run sysinfo; setenv bootargs console=${console},${baudrate} root=${usbroot} ${bootargs_sysinfo} ${opt_args} 
-usbboot=echo Booting from USB ...; run usbargs; if test ${boot_fdt} = yes || test ${boot_fdt} = try; then if run usbloadfdt; then bootz ${loadaddr} - ${fdt_addr}; else if test ${boot_fdt} = try; then bootz; else echo WARN: Cannot load the DT; fi; fi; else bootz; fi; 
+splashsource=mmc_fs
+sysinfo=setenv bootargs_sysinfo board_name=${board_name} board_variant=${variant_key}-${feature_key} serial_number=${serial_number} uboot_version=<${uboot_version}>
+uboot_version=U-Boot 2018.03 (Oct 10 2019 - 12:21:18 +0000)
```
Comment (shanx) [inline env/u-boot_env.txt:74]: Should this version be updated? I read in the PR: ```Update UBoot to MSC version 2020.01```

File: env/u-boot_env.txt [hunk 2]
```diff
@@ -1,70 +1,85 @@
...
 emmcextpart=1
-emmcloadenv=echo Loading environment (${envfile}) from eMMC ...; ext4load mmc ${emmcdev}:${emmcextpart} ${loadaddr} ${envfile}; 
-emmcloadfdt=echo Loading FDT image (${imageroot}/${fdtfile}) from eMMC ...; ext2load mmc ${emmcdev}:${emmcextpart} ${fdt_addr} ${fdtfile}; 
-emmcloadimage=echo Loading linux image (${imageroot}/${image}) from eMMC ...; ext4load mmc ${emmcdev}:${emmcextpart} ${loadaddr} ${image}; 
+emmcloadenv=echo Loading environment (${envfile}) from eMMC ...; ext4load mmc ${emmcdev}:${emmcextpart} ${loadaddr} ${envfile};
+emmcloadfdt=echo Loading FDT image (${imageroot}/${fdtfile}) from eMMC ...; ext2load mmc ${emmcdev}:${emmcextpart} ${fdt_addr} ${fdtfile};
+emmcloadimage=echo Loading linux image (${imageroot}/${image}) from eMMC ...; ext4load mmc ${emmcdev}:${emmcextpart} ${loadaddr} ${image};
 emmcroot=/dev/mmcblk2p2 rootwait rw
 envfile=uEnv.txt
+ethprime=FEC
```
Comment (shanx) [inline env/u-boot_env.txt:23]: Is it possible to add documentation for this value? What does FEC mean?

File: env/u-boot_env.txt [hunk 3]
```diff
@@ -1,70 +1,85 @@
...
 fdt_addr=0x18000000
 fdt_high=0xffffffff
 fdtfile=msc-sm2-imx6dl-ultimain4.2.dtb
 fdtroot=boot
+feature_key=013
```
Comment (shanx) [inline env/u-boot_env.txt:28]: Is it possible to add documentation for this value? What does 013 mean here?

File: env/u-boot_env.txt [hunk 4]
```diff
@@ -1,70 +1,85 @@
...
 net_boot=if test -n ${serverip} ; then dhcp ;if test -n ${nfsrootpath} ; then nfs ${fdt_addr} ${serverip}:${nfsrootpath}/boot/${fdtfile}; fdt addr ${fdt_addr}; fdt resize;nfs ${loadaddr} ${serverip}:${nfsrootpath}/boot/${image}; run nfsargs; bootz ${loadaddr} - ${fdt_addr};else echo ERR: nfsrootpath not set; false; fi;else echo ERR: serverip not set; false; fi;
-nfsargs=run sysinfo; setenv bootargs console=${console},${baudrate} nfsroot=${nfsrootpath} rootwait rootfstype=nfs ip=${ipaddr}:${serverip}:${gateway}:${netmask} ${opt_args} 
+nfsargs=run sysinfo; setenv bootargs console=${console},${baudrate} nfsroot=${nfsrootpath} rootwait rootfstype=nfs ip=${ipaddr}:${serverip}:${gateway}:${netmask} ${opt_args}
 opt_args=consoleblank=0 earlyprintk pci=nomsi
-revision=N/A
-serial_number=1007481365
-spi_import_logo=mmc dev ${mmcdev}; if mmc rescan; then echo Loading logo image (${splashfile}) from mmc ...; if ext4load mmc ${mmcdev}:${mmcextpart} $loadaddr ${splashfile}; then sf probe; echo Updating logo partition ...; sf update $loadaddr logo $filesize; setenv splashsource sf; saveenv; echo succeeded; else echo failed; fi; fi;
+revision=A0
```
Comment (shanx) [inline env/u-boot_env.txt:51]: This is our own internal revision number?

File: env/u-boot_env.txt [hunk 5]
```diff
@@ -1,70 +1,85 @@
...
 usbextpart=2
 usbfatpart=1
-usbloadenv=echo Loading environment (${envfile}) from USB ...; fatload usb ${usbdev}:${uenvpart} ${loadaddr} ${imageroot}/${envfile}; 
-usbloadfdt=echo Loading FDT image (${fdtroot}/${fdtfile}) from USB ...; ext4load usb ${usbdev}:${usbextpart} ${fdt_addr} ${fdtroot}/${fdtfile}; 
-usbloadimage=echo Loading linux image (${imageroot}/${image}) from USB ...; ext4load usb ${usbdev}:${usbextpart} ${loadaddr} ${imageroot}/${image}; 
+usbloadenv=echo Loading environment (${envfile}) from USB ...; fatload usb ${usbdev}:${uenvpart} ${loadaddr} ${imageroot}/${envfile};
+usbloadfdt=echo Loading FDT image (${fdtroot}/${fdtfile}) from USB ...; ext4load usb ${usbdev}:${usbextpart} ${fdt_addr} ${fdtroot}/${fdtfile};
+usbloadimage=echo Loading linux image (${imageroot}/${image}) from USB ...; ext4load usb ${usbdev}:${usbextpart} ${loadaddr} ${imageroot}/${image};
 usbroot=/dev/sda2 rootwait rw
+variant_key=03E
```
Comment (shanx) [inline env/u-boot_env.txt:85]: what does this variable mean?

**Review by shanx — COMMENTED** [review]
Looks good. Some questions, see PR

File: env/u-boot_env.txt (hunk 1 above)
Comment (alexborro) [reply env/u-boot_env.txt:74]: This variable is completely irrelevant, it is not used. I will remove it.

File: env/u-boot_env.txt (hunk 2 above)
Comment (alexborro) [reply env/u-boot_env.txt:23]: I have no idea, this is probably some variable uboot sends to the kernel, but I'm not sure what driver uses it.

File: env/u-boot_env.txt (hunk 3 above)
Comment (alexborro) [reply env/u-boot_env.txt:28]: This is something internal to MSC, I also don't know that it means.

File: env/u-boot_env.txt (hunk 4 above)
Comment (alexborro) [reply env/u-boot_env.txt:51]: Again, this is related to some MSC Revision. All this configuration came ready from MSC.

File: env/u-boot_env.txt (hunk 5 above)
Comment (alexborro) [reply env/u-boot_env.txt:85]: This is the module variant. The "old" one is 03A, this new module is the 03E. Again, this parameter is not used by our kernel.

File: env/u-boot_env.txt (hunk 4 above)
Comment (shanx) [reply env/u-boot_env.txt:51]: Can we reference the source of these values then?

**Review by bryanfoley — APPROVED** [review]

**Review by shanx — DISMISSED** [review]
Let's add least add some documentation about the MSC variables and where they came from.

File: env/u-boot_env.txt (hunk 4 above)
Comment (alexborro) [reply env/u-boot_env.txt:51]: The source is the original (private) MSC customized U-Boot version. It is cloned in out repo: git@github.com:Ultimaker/u-boot-denx-msc.git (it is a submodule of the um-u-boot repo).
About **all** variables working well in the current IMX6, I'm not sure, but most of those customized variables doesnt play any role in our kernel.

**Review by compihu — APPROVED** [review]

File: env/u-boot_env.txt (hunk 4 above)
Comment (bryanfoley) [reply env/u-boot_env.txt:51]: @shanx Would the above explanation of where the code changes were coming from satisfy as the minimal required documentation if added to the README.md of this branch? The code in that repo is also quite light on explanation of variables but I cannot find a comprehensive documentation.

**Review by robinmdh — APPROVED** [review]

File: env/u-boot_env.txt (hunk 4 above)
Comment (shanx) [reply env/u-boot_env.txt:51]: Yeah that would be fine. I just want to make sure we leave a trace in such a way that should for example Alex leave the company it some point, someone else can pick up and can trace all this back

Comment (bryanfoley) [reply env/u-boot_env.txt:51]: @shanx a very short README.md has been added which explains where the underlying proprietary code comes from and contains a short description of how to build. The repo we submodule can be used to discover deeper what code changes were recently made by MSC to introduce support for the imx6 rev 03E. It is lacking in detail as I do not know enough about um-u-boot to explain what it truly does.

**Pull request merged**